File: c#/money-problem.Tests/MoneyCalculatorShould.cs
using FluentAssertions;
using money_problem.Domain;
using Xunit;
using static money_problem.Domain.Currency;

namespace money_problem.Tests
{
    public class MoneyShould
    {
        [Fact(DisplayName = "10 EUR x 2 = 20 EUR")]
        public void MultiplyInEuros()
        {
            MoneyCalculator
                .Times(10, EUR, 2)
                .Should()
                .Be(20d);
        }

        [Fact(DisplayName = "4002 KRW / 4 = 1000.5 KRW")]
        public void DivideInKoreanWons()
        {
            MoneyCalculator
                .Divide(4002, KRW, 4)
                .Should()
                .Be(1000.5d);
        }
    }
}

File: scala/src/test/scala/domain/MoneyCalculatorShould.scala
package domain

import domain.Currency._
import org.scalatest.funsuite.AnyFunSuite

class MoneyCalculatorShould extends AnyFunSuite {
  test("5 USD + 10 USD = 15 USD") {
    assert(
      MoneyCalculator
        .add(5, USD, 10) === 15
    )
  }

  test("10 EUR x 2 = 20 EUR") {
    assert(
      MoneyCalculator
        .times(10, EUR, 2) === 20
    )
  }

  test("4002 KRW / 4 = 1000.5 KRW") {
    assert(
      MoneyCalculator
        .divide(4002, KRW, 4) === 1000.5
    )
  }
}
